shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -268,15 +269,13 @@ private void load(AlluxioURI filePath, int replication)
       throws IOException, AlluxioException {
     URIStatus status = mFileSystem.getStatus(filePath);
     if (status.isFolder()) {
-      List<URIStatus> statuses = mFileSystem.listStatus(filePath);
-      for (URIStatus uriStatus : statuses) {
-        if (uriStatus.isFolder()) {
-          AlluxioURI subPath = new AlluxioURI(uriStatus.getPath());
-          load(subPath, replication);
-        } else {
-          addJob(uriStatus, replication);
-        }
-      }
+      ListStatusPOptions options = ListStatusPOptions.newBuilder().setRecursive(true).build();
```

Comment:
There is a tradeoff for this recursive listing. We could be locking more of the tree for longer with the recursive listing (especially if the sync interval was triggered)

Reply:
but it is more 'correct' as you are guaranteed to have a 'snapshot' of a dir, versus dealing with contentions.
also this can save RPC cost on master side which IMO can offset the locking overhead.